src/main/java/com/cloudbees/jenkins/plugins/bitbucket/client/BitbucketCloudApiClient.java

```diff
@@ -245,6 +245,12 @@ public String getRepositoryUri(@NonNull BitbucketRepositoryType type,
             case GIT:
                 switch (protocol) {
                     case HTTP:
+                        if(authenticator != null ) {
+                            String username = authenticator.getUserUri();
+                            if (!username.isEmpty()) {
+                                return "https://" + username + "@bitbucket.org/" + owner + "/" + repository + ".git";
```

Comment:
not sure that would work as username would return
```
    public String getUserUri() {
        return "x-token-auth:{" + token.getToken() + "}";
    }
```

Reply:
I guess it does according to the API: https://confluence.atlassian.com/bitbucket/oauth-on-bitbucket-cloud-238027431.html

Reply:
Yes. I introduced getUserUri() member method in the authenticator in order to use the retrieved token according to the documentation you linked